src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +159,175 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
     }
 }
 
+void IInferRequestInternal::SetBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs) {
+    bool all_memory = std::all_of(blobs.begin(), blobs.end(), [](const Blob::Ptr& item) {
+        return item && item->is<MemoryBlob>() && !item->is<RemoteBlob>();
+    });
+    OPENVINO_ASSERT(all_memory,
+                    "set_input_tensors/set_tensors error. Default implementation support only local memory tensors");
+
+    if (blobs.size() == 1) {
```

Comment:
should not check this be moved to the begging of the method before check for `all_memory`?